File: sach/de3378.cpp
//DE3378.CPP - Tran Huu Nam - hsgtin.vn - 2024
#include <bits/stdc++.h>
using namespace std;
int main()
{
    int n;
    string s;
    map<string, int> h;
    cin>>n;
    while (n--) {
        cin>>s;
        if (h[s]==0) cout<<s<<endl;
        else cout<<s<<h[s]<<endl;
        h[s]++;
    }
    return 0;
}

File: sach/de3378.py
//DE3378.PY - Tran Huu Nam - hsgtin.vn - 2024
import sys
from collections import defaultdict
def main():
    n = int(sys.stdin.readline().strip())
    h = defaultdict(int)
    for _ in range(n):
        s = sys.stdin.readline().strip()
        if h[s] == 0:
            print(s)
        else:
            print(f"{s}{h[s]}")
        h[s] += 1
if __name__ == "__main__":
    main()

